src/main/java/io/jenkins/plugins/pipelinegraphview/PipelineGraphDisplayURLProvider.java

```diff
@@ -24,6 +25,30 @@ public String getName() {
     @Override
     @NonNull
     public String getRunURL(Run<?, ?> run) {
-        return getRoot() + Util.encode(run.getUrl()) + "pipeline-graph";
+        if (run instanceof WorkflowRun) {
+            return DisplayURLProvider.getDefault().getRunURL(run) + "pipeline-graph";
+        }
+        return DisplayURLProvider.getDefault().getRunURL(run);
```

Comment:
Otherwise, for non Pipelines, we navigate to `/pipeline-graph`, which then 404s and sends us back to the main build page, which is an unnecessary redirect.